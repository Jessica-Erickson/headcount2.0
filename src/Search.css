.Search {
  display: flex;
  justify-content: flex-end;
}

input[type='text'] {
  border: thin solid lightgray;
  border-radius: .5rem;
  box-sizing: border-box;
  font-size: 1.5rem;
  height: 2.5rem;
  margin: 1rem;
  padding: .5rem;
  width: 20rem;
}