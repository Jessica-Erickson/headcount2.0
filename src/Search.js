import React from 'react';
import './Search.css';
import PropTypes from 'prop-types';

const Search = ({ searchValue , handleSearch }) => {
  return (
    <div className='Search'>
      <input type='text' value={searchValue} onChange={handleSearch} placeholder='Search' />
    </div>
  )
}

Search.propTypes = {
  searchValue: PropTypes.string.isRequired,
  handleSearch: PropTypes.func.isRequired
}

export default Search;
