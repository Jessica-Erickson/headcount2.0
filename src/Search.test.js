import React from 'react';
import ReactDOM from 'react-dom';
import Search from './Search';
import {shallow} from 'enzyme';

describe('Search', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Search 
                      searchValue={''}
                      handleSearch={() => {}} />, 
                    div);
  });

  it('should match the snapshot', () => {
    const wrapper = shallow(<Search 
                              searchValue={''}
                              handleSearch={() => {}} />);

    expect(wrapper).toMatchSnapshot()
  });
});
