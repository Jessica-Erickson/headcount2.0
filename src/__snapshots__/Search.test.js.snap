// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search should match the snapshot 1`] = `
<div
  className="Search"
>
  <input
    onChange={[Function]}
    placeholder="Search"
    type="text"
    value=""
  />
</div>
`;
